DigitalLearningSolutions.Data/Models/CustomPrompts/CustomPrompt.cs

```diff
@@ -2,18 +2,19 @@
 {
     using System;
     using System.Collections.Generic;
+    using DigitalLearningSolutions.Data.Enums;
 
     public class CustomPrompt
     {
-        public CustomPrompt(int customPromptNumber, string text, string? options, bool mandatory)
+        public CustomPrompt(RegistrationField registrationField, string text, string? options, bool mandatory)
         {
-            CustomPromptNumber = customPromptNumber;
+            RegistrationField = registrationField;
             CustomPromptText = text;
             Options = SplitOptionsString(options);
             Mandatory = mandatory;
         }
 
-        public int CustomPromptNumber { get; set; }
+        public RegistrationField RegistrationField { get; set; }

```

Comment:
I've not had a thorough look through this commit, but I think I am somewhat opposed to this refactor. The CustomPrompt class is also used for the data in CourseAdminFields as they follow a similar format, but those technically aren't "RegistrationFields" and don't have a corresponding linked to field. There is also a maximum of 3 of them. So this refactor might need something more pernickety to it.

Reply:
Here's a separate MR for the changes from that commit: https://github.com/TechnologyEnhancedLearning/DLSV2/pull/971

I've also created a different model for the admin fields: CoursePrompt. CustomPrompt now has a RegistrationField instead of a CustomPromptNumber, but CoursePrompt is structured exactly as CustomPrompt was previously.